src/main/java/com/tabnine/binary/fetch/BinaryVersionFetcher.java

```diff
@@ -27,6 +28,12 @@ public BinaryVersionFetcher(LocalBinaryVersions localBinaryVersions, BinaryRemot
      * @throws SecurityException, NoExistingBinaryException if something went wrong
      */
     public String fetchBinary() throws NoValidBinaryToRunException {
+        Optional<BinaryVersion> bootstrappedVersion = BootstrapperSupport.bootstrap_version(localBinaryVersions, binaryRemoteSource, binaryDownloader);
+        if (bootstrappedVersion.isPresent()) {
+            Logger.getInstance(getClass()).info(format("found local bootstrapped version %s", bootstrappedVersion.get()));
+            return bootstrappedVersion.get().getVersionFullPath();
+        }
+        Logger.getInstance(getClass()).warn("couldn't get bootstrapped version. fallback to legacy code!");
```

Comment:
extract to a method in a different class

Reply:
it is in a different class